bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -0,0 +1,265 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.senechome.internal.json.SenecHomeResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SenecHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Steven Schwarznau - Initial contribution
+ */
+@NonNullByDefault
+public class SenecHomeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SenecHomeHandler.class);
+    private final static String VALUE_TYPE_INT = "u3";
+    private final static String VALUE_TYPE_UNSIGNED_INT = "u8";
+    private final static String VALUE_TYPE_FLOAT = "fl";
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable PowerLimitationStatusDTO limitationStatus = null;
+    private @Nullable SenecHomeApi senecHomeApi;
+    private Gson gson;
+    private SenecHomeApiFactory apiFactory;
+    private SenecHomeConfigurationDTO config = new SenecHomeConfigurationDTO();
+
+    public SenecHomeHandler(Thing thing, Gson gson, SenecHomeApiFactory apiFactory) {
+        super(thing);
+        this.gson = gson;
+        this.apiFactory = apiFactory;
+    }
+
+    @Override
+    public void handleRemoval() {
+        stopJobIfRunning();
+        super.handleRemoval();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        /* not implemented - binding only supports reading */
+    }
+
+    @Override
+    public void dispose() {
+        stopJobIfRunning();
+        super.dispose();
+    }
+
+    protected void stopJobIfRunning() {
+    	final ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SenecHomeConfigurationDTO.class);
+        senecHomeApi = apiFactory.getHttpApi(config, gson);
+        refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void refresh() {
+        try {
+            SenecHomeResponse response = senecHomeApi.getStatistics();
+            logger.trace("received {}", response);
+
+            BigDecimal pvLimitation = new BigDecimal(100).subtract(getSenecValue(response.limitation.powerLimitation))
+                    .setScale(0, RoundingMode.HALF_UP);
+
+            Channel channelLimitation = getThing().getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_POWER_LIMITATION);
+            updateState(channelLimitation.getUID(),
+            		new QuantityType<Dimensionless>(pvLimitation, SmartHomeUnits.PERCENT));
```

Comment:
Why do you need the channel at all? `updateState` allows the `channelId`, which is exactly what you pass to `getChannel`. 
```suggestion
            updateState(SenecHomeBindingConstants.CHANNEL_SENEC_POWER_LIMITATION,
            		new QuantityType<Dimensionless>(pvLimitation, SmartHomeUnits.PERCENT));
```